### PR TITLE
Update client SDK

### PR DESCRIPTION
**Pull request by gracez72**
**Main files to check**:
- `adaptivecards-templating-client/swagger.yaml`: client SDK definition updated to backend changes in this [PR](https://github.com/microsoft/adaptivecards-templates/pull/84)
- `adaptivecards-templating-service/src/TemplateServiceClient.ts`: fix tags to be in query rather than body of get request

**Testing**
The client SDK changes have been tested locally.

### CODE REVIEW

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 1]
```diff
@@ -678,13 +1154,14 @@ export class TemplateApi {
...
-     * @param body Template object to be added
+     * Returns a single template
+     * @summary Find template by id
+     * @param templateId ID of template to return
+     * @param isPublished Query based on if template is published
+     * @param version Version of template to return
      * @param {*} [options] Override http request options.
      */
-    public templateTemplateIdPost (templateId: string, body: TemplateJSON, options: any = {}) : Promise<{ response: http.IncomingMessage; body?: any;  }> {
+    public templateById (templateId: string, isPublished?: boolean, version?: string, options: any = {}) : Promise<{ response: http.IncomingMessage; body: TemplateList;  }> {
```
Comment (guptaharsh25) [inline private-templates-service/adaptivecards-templating-client/typescript/api.ts]: I might be missing something here, but why would we want to get a templateByID and provide a param for isPublished? If we want a template by ID then I don't think we need isPublished

Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: It's for if you want to get only the published/unpublished versions of a template

Comment (guptaharsh25) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: ohhh okay cool

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 2]
```diff
@@ -227,8 +375,8 @@ export class Template {
             "type": "string"
         },
         {
-            "name": "isPublished",
-            "baseName": "isPublished",
+            "name": "isLive",
```
Comment (daniellemott) [inline private-templates-service/adaptivecards-templating-client/typescript/api.ts]: Is there a different meaning behind "isLive" and "isPublished"?

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 3]
```diff
@@ -551,36 +1041,16 @@ export class TemplateApi {
...
     }
     /**
-     * Returns all public templates and owned templates
-     * @summary Find all templates
-     * @param isPublished Query based on if template is published
-     * @param name Name of template to query for
-     * @param version Version of template
-     * @param owned Display only the templates owned by the user
+     * 
+     * @summary Get recently viewed, edited templates, and recent tags.
```
[the remark this answers is not in the document]
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: It's a list of tags that the logged in user has recently added to their templates

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts (hunk 2 above)
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: Whoops, isLive should be the only parameter under Template, I'll fix that!

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 4]
```diff
@@ -245,20 +393,79 @@ export class Template {
...
+    }
+}
+
 export class TemplateInstance {
+    'id'?: string;
     'json'?: string;
     'version'?: string;
+    'publishedAt'?: string;
+    'state'?: TemplateInstance.StateEnum;
+    'isShareable'?: boolean;
```
[the remark this answers is not in the document]
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: The variable names are taken from the server model, I think isShareable is okay for now?

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts (hunk 2 above)
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: isLive means the same thing as isPublished, isLive is one of the fields returned by the templates after adding the different states (live, draft, deprecated)

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 5]
```diff
@@ -461,45 +807,188 @@ export class TemplateApi {
...
-            if (basePath) {
-                this.basePath = basePath;
-            }
-        } else {
-            if (basePathOrUsername) {
-                this.basePath = basePathOrUsername
-            }
+    constructor(basePath?: string);
+    constructor(basePathOrUsername: string, password?: string, basePath?: string) {
+        if (password) {
```
Comment (Jackbk) [inline private-templates-service/adaptivecards-templating-client/typescript/api.ts]: if(password && basepath)

Comment (Jackbk) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: wait is this file generate?

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 6]
```diff
@@ -535,12 +1026,11 @@ export class TemplateApi {
                 localVarRequestOptions.form = localVarFormParams;
             }
         }
-        return new Promise<{ response: http.IncomingMessage; body: TemplateList;  }>((resolve, reject) => {
+        return new Promise<{ response: http.IncomingMessage; body?: any;  }>((resolve, reject) => {
```
Comment (Jackbk) [inline private-templates-service/adaptivecards-templating-client/typescript/api.ts]: intentional change?

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts (hunk 5 above)
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: Yep, this file was generated from the yaml definition!

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts (hunk 6 above)
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: the api.ts file was generated, but I'll make a note to look through the generated file and fix it up once the endpoint definitions are set!

**Comment by daniellemott** [comment]
> lgtm

Ran locally.
lgtm pt 2

**Pull request merged**